File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []
        qu=deque([root])
        res=[]
        
        while qu:
            lev_size=len(qu)
            for i in range(lev_size):
                cur=qu.popleft()
                if (i == (lev_size-1)):
                    res.append(cur.val)
                
                if cur.left:
                    qu.append(cur.left)
                if cur.right:
                    qu.append(cur.right)
        return res
        

File: Data Structures & Algorithms/binary-tree-diameter/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int max=0;
    public int ht (TreeNode root){
        if(root==null) return 0;
       int l= ht(root.left);
       int r= ht(root.right);
       max=Math.max(max,l+r);
       return 1 + Math.max(l,r);
    }
    public int diameterOfBinaryTree(TreeNode root) {
        ht(root);
        return max;
    }
}
